README.md

```diff
@@ -24,247 +21,310 @@
 - [Usage](#usage)
 
 ## Intro
-This chat app allows the user to converse with Gemini and use it as an intelligent, personal AI assistant.
-Two modes of text-only chat are currently available in this app, non-streaming and streaming.
 
-In non-streaming mode, a response is returned after Gemini completes the entire text generation process.
+This example app lets the user chat with Gemini and use it as a personal AI
+assistant. The app supports text-only chat in two modes: non-streaming and
+streaming.
 
-Streaming mode uses Gemini's streaming capability to achieve faster interactions.
+In non-streaming mode, a response is returned after Gemini completes the entire
+text generation process.
 
-### Frontend 
-The client for this app is written using [React](https://react.dev/) and served using [Vite](https://github.com/vitejs/vite).
+Streaming mode uses Gemini's streaming capability to achieve faster
+interactions.
+
+### Frontend
+
+The client for this app is written using [React](https://react.dev/) and served
+using [Vite](https://github.com/vitejs/vite).
 
 ### Backend
-The app currently has 3 different backend servers that the user can choose from, [Flask](https://flask.palletsprojects.com/en/3.0.x/quickstart/), [Node.js](https://Node.js.org/en) or [Go](https://go.dev)
+
+The example provides three implementations of the backend server:
+
+* A Python [Flask](https://flask.palletsprojects.com/en/3.0.x/quickstart/)
+  server, which demonstrates the
+  [Gemini API Python SDK](https://github.com/google-gemini/generative-ai-python)
+* A Node.js server, which demonstrates the
+  [Gemini API JavaScript SDK](https://github.com/google-gemini/generative-ai-js).
+* A Go server, which demonstrates the
+  [Gemini API Go SDK](https://github.com/google/generative-ai-go) 
 
 ## API documentation
```

Comment:
Just a thought - would this be better if they were moved after the setup steps? Seems like the value of the app is understanding & running, not necessarily providing a layer of indirection in front of the real API.